server/events/vcs/github_client.go

```diff
@@ -430,16 +481,106 @@ func (g *GithubClient) UpdateStatus(ctx context.Context, request types.UpdateSta
 }
 
 // [WENGINES-4643] TODO: Move the checks implementation to UpdateStatus once github checks is stable
-// UpdateChecksStatus updates the status check
 func (g *GithubClient) UpdateChecksStatus(ctx context.Context, request types.UpdateStatusRequest) error {
-	// TODO: Implement updating github checks
-	// - Get all checkruns for this SHA
-	// - Match the UpdateReqIdentifier with the check run. If it exists, update the checkrun. If it does not, create a new check run.
+	checkRuns, err := g.GetRepoChecksForRef(request.Repo, request.Ref)
+	if err != nil {
+		return err
+	}
 
-	// Checks uses Status and Conlusion. Need to map models.CommitStatus to Status and Conclusion
-	// Status -> queued, in_progress, completed
-	// Conclusion -> failure, neutral, cancelled, timed_out, or action_required. (Optional. Required if you provide a status of "completed".)
-	return nil
+	for _, checkRun := range checkRuns {
```

Comment:
Can you have more than 1 checkrun per status name?

Reply:
By status name, do you mean command name? 
We create a check run for the following combination `{commit-sha, project-name, command}` which means a new check run is created if any of the three variables change. However, we update the same check run when rerunning a command. So, in my opinion, I don't think we can have more than 1 check run per status name.